src/classes/Application/Traits/Admin/Wizard.php

```diff
@@ -1,260 +1,295 @@
+     * @var string
+     */
     protected $sessionID;
-    
-   /**
-    * @var string
-    */
+
+    /**
+     * @var string
+     */
     protected $classBase;
 
-   /**
-    * @var array
-    */
+    /**
+     * @var array
+     */
     protected $sessionData = array();
-    
-   /**
-    * @var array<string,Application_Admin_Wizard_Step>
-    */
+
+    /**
+     * @var array<string,Application_Admin_Wizard_Step>
+     */
     protected $steps = array();
-    
-   /**
-    * @var string
-    */
+
+    /**
+     * @var string
+     */
     protected $initialStepName;
-    
-   /**
-    * @var Application_Admin_Wizard_Step
-    */
+
+    /**
+     * @var Application_Admin_Wizard_Step
+     */
     protected $activeStep;
-    
-   /**
-    * Set when a specific step has been requested via the request.
-    * @var Application_Admin_Wizard_Step|NULL
-    */
+
+    /**
+     * Set when a specific step has been requested via the request.
+     * @var Application_Admin_Wizard_Step|NULL
+     */
     protected $requestedStep;
 
-   /**
-    * @var string
-    */
+    /**
+     * @var string
+     */
     protected $settingPrefix = '';
-    
+
+    /**
+     * @var InvalidationHandler
+     */
+    protected InvalidationHandler $invalidationHandler;
+
+    /**
+     * @var bool
+     */
+    protected bool $hasPrevTransaction = true;
+
     abstract public function getWizardID() : string;
-    
+
     abstract public function getClassBase() : string;
-    
+
     abstract public function getSuccessMessage() : string;
 
     protected function getTemplateName() : string
     {
         return 'content/wizard';
     }
-    
+
     public function initWizard() : void
     {
         $this->wizardID = $this->getWizardID();
         $this->classBase = $this->getClassBase();
-        
-        // create or get the wizard session ID: this is used to 
+
+        // create or get the wizard session ID: this is used to
         // identify a specific wizard session, and enables the
         // same user to use several parallel wizards. The ID is
         // included in all URLs and forms.
         $this->sessionID = $this->request->getParam('wizard');
-        
-        if(empty($this->sessionID)) 
+
+        if (empty($this->sessionID))
         {
-            $this->sessionID = 'WZ'.crc32(microtime(true).'-wizard-'.$this->user->getID());
+            $this->sessionID = 'WZ' . crc32(microtime(true) . '-wizard-' . $this->user->getID());
 
             $this->log(sprintf('Generated the wizard ID [%s].', $this->sessionID));
-            
+
             $this->sessionData = array(
                 'sessionID' => $this->sessionID,
                 'userID' => $this->user->getID(),
                 'lastActive' => time()
             );
-            
+            $this->invalidationHandler = new InvalidationHandler();
+            $this->invalidationHandler->setIsInvalidated(false);
+            $this->sessionData['invalidationHandler'] = $this->invalidationHandler;
```

Comment:
We have the `setWizardSetting()` and `getWizardSetting()` methods, it's better to use them instead of writing into the property directly. The methods support using objects.